### ISSUE TITLE
com.facebook.android:facebook-common version 17.0.1 has Device and Network Abuse policy issues that will block publishing of your app to Play Console in the future

### ISSUE THREAD

**Issue by andrea7887** [facebook/facebook-android-sdk#1262]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [Github issues](https://github.com/facebook/facebook-android-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Java version

17

### Android version

34

### Android SDK version

com.facebook.android:facebook-share:17.0.1

### Installation platform & version

gradle

### Package

Core & AppEvents

### Goals

I would like to submit an app to google play using the share library just to share a link to facebook

### Expected results

No errors on android studio

### Actual results

com.facebook.android:facebook-common version 17.0.1 has Device and Network Abuse policy issues that will block publishing of your app to Play Console in the future

### Code samples & details

```java
// INSERT YOUR CODE HERE
```

**Comment by angiep125** [comment]
Hi @andrea7887 please consider switching over to using our Intent API share integrations
https://developers.facebook.com/docs/sharing/sharing-to-stories/android-developers

Sharing to feed can be done using a SEND intent. We also have Reels 

https://developers.facebook.com/docs/android/sharing-to-reels-facebook/

These flows are actively maintained by our teams. This should also allow you to more flexibility in the UI of the facebook sharing buttons on your application.

Please do let us know if you have any issues integrating.

Taken a look at this previous issue for a solution
https://github.com/facebook/facebook-android-sdk/issues/1081#issuecomment-1159241137

Remember to use FileProvider instead of Uri.parse as our docs.

**Comment by angiep125** [comment]
https://github.com/paulocoutinhox/android-share-story
See paulo's example

**Comment by andrea7887** [comment]
Hi, thank you a lot for your reply. For the SEND intent to share to the feed is ok something like:

                val sendIntent = Intent()
                sendIntent.action = Intent.ACTION_SEND
                sendIntent.putExtra(Intent.EXTRA_TEXT, "$title $link")
                sendIntent.type = "text/plain"
                sendIntent.setPackage("com.facebook.katana")
                val packageManager = requireActivity().packageManager
                if (sendIntent.resolveActivity(packageManager) != null) {
                    startActivity(sendIntent)
                    return
                }}

**Comment by angiep125** [comment]
Yes! that is the correct package name

**Comment by cemtufekci** [comment]
that warning still exists on 17.0.2

**Comment by Husnainmirza0071** [comment]
Thanks for everything

**Comment by Andrew0000** [comment]
Is there a solution? I still have the error in 17.0.2:

![image](https://github.com/user-attachments/assets/c0d2ebe9-6347-4d98-81db-e73b9d73067a)

**Comment by ronenfe** [comment]
com.facebook.android:facebook-common:17.0.2
com.facebook.android:facebook-common version 17.0.2 has Device and Network Abuse policy issues that will block publishing of your app to Play Console in the future

Me too, I think I can't use facebook campaigns for the app without it.

**Comment by angiep125** [comment]
please consider switching over to using our Intent API share integrations
https://developers.facebook.com/docs/sharing/sharing-to-stories/android-developers

Sharing to feed can be done using a SEND intent. We also have Reels

https://developers.facebook.com/docs/android/sharing-to-reels-facebook/

These flows are actively maintained by our teams. This should also allow you to more flexibility in the UI of the facebook sharing buttons on your application.

Please do let us know if you have any issues integrating.

Taken a look at this previous issue for a solution
https://github.com/facebook/facebook-android-sdk/issues/1081#issuecomment-1159241137

Remember to use FileProvider instead of Uri.parse as our docs.

https://github.com/paulocoutinhox/android-share-story
See paulo's example

**Comment by ronenfe** [comment]
I don't use sharing, I use it for advertising on Facebook.

**Comment by ronenfe** [comment]
I don't use sharing, I use it for advertising on Facebook.

**Comment by ronenfe** [comment]
I removed the dependency, it seems to build, maybe I will not be able to advertise anymore on Facebook or get the statistics, but it's ok. I don't need to use it anymore.

**Comment by Andrew0000** [comment]
In my case I need the SDK for advertising too. Not for sharing.

**Comment by Sar777** [comment]
bump, any news? it's valid for 17.0.2 as well

**Comment by adrianapele** [comment]
I use "com.facebook.android:facebook-login" and have the same issue. excluding the
"com.facebook.android:facebook-common"  from the above lib results in compile errors.

**Comment by wiggisser** [comment]
@angiep125 We only use `com.facebook.android:facebook-login:17.0.2` and observe the same issue. Is there a suggested fix/workaround for this. Because the official docs on "Facebook Login" still recommend to use `com.facebook.android:facebook-login` dependency.

**Comment by cemtufekci** [comment]
yeah, excluding common library should not be the expected solution. we are still waiting this warning to be fixed.

**Comment by vivian688** [comment]
Has the latest version been repaired?@angiep125

**Comment by cemtufekci** [comment]
after updating to ladybug 2024.2.2 stable version, the warning has gone now
